### PR TITLE
[Feature] Add Next JS

### PR DESCRIPTION
**Pull request by gsavchenko**
Loom: https://www.loom.com/share/6009a96808a446c5abb1b5b065a36dd0

These changes are separating out our Frontend and Backend code. Now we're using NextJS to handle our frontend/routing. The consequences of this are that we are not going to be relying on the views/routing features given to us by Pheonix. We should also follow up and remove the existing implementation of our components on the BE and packages related to frontend work.

If you want to run this yourself:
from the root umbrella app run:
docker compose up -d (this runs our postgres DB)
mix phx.server (this starts our BE server)

from kur_kards_frontend:
yarn dev (this starts the frontend server)

### CODE REVIEW

File: kur_kards_frontend/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    setMessage(message);
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const name = event.target.value;
+    setName(name);
+  };
+
+  return (
+    <>
```
Comment (vicovaro) [inline kur_kards_frontend/pages/index.tsx:53]: React Fragment is recommended

File: kur_kards_frontend/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            </div>
+          );
+        })}
+      </div>
+
+      <div className="row form-group">
+        <div className="col-md-3">
+          <input
+            type="text"
+            id="name"
```
Comment (vicovaro) [inline kur_kards_frontend/pages/index.tsx:69]: I don't know if you really need it, or you do it out of tradition.

File: kur_kards_frontend/styles/Home.module.css [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  height: 1em;
+  margin-left: 0.5rem;
+}
+
+@media (max-width: 600px) {
+  .grid {
+    width: 100%;
+    flex-direction: column;
+  }
+}
```
Comment (vicovaro) [inline kur_kards_frontend/styles/Home.module.css:116]: vanilla CSS lol

**Review by vicovaro — CHANGES_REQUESTED** [review]
just look at comments, otherwise lgtm

File: kur_kards_frontend/pages/index.tsx (hunk 2 above)
Comment (gsavchenko) [reply kur_kards_frontend/pages/index.tsx:69]: totally agree with your comments; however, this is just for POC purposes and will be totally replaced in the future.

File: kur_kards_frontend/styles/Home.module.css (hunk 3 above)
Comment (gsavchenko) [reply kur_kards_frontend/styles/Home.module.css:116]: same here, this is just to see that we can run pheonix and nextjs together

File: kur_kards_frontend/pages/index.tsx (hunk 1 above)
Comment (gsavchenko) [reply kur_kards_frontend/pages/index.tsx:53]: this sounds like an opinion, why use the long form? i like the short hand better. either way we should add an eslint rule for one or the other and enforce it automatically.

**Review by vicovaro — APPROVED** [review]
LGTM

**Pull request merged**